### PR TITLE
VP-4251: Implement basic password hashing extensibility

### PR DESCRIPTION
**Pull request by akak1977**

### CODE REVIEW

File: src/VirtoCommerce.Platform.Core/Security/IUserPasswordHasher.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+using Microsoft.AspNetCore.Identity;
+
+namespace VirtoCommerce.Platform.Core.Security
+{
+    /// <summary>
+    /// Basic interface for platform password hashers
+    /// </summary>
+    public interface IUserPasswordHasher
```
Comment (tatarincev) [inline src/VirtoCommerce.Platform.Core/Security/IUserPasswordHasher.cs:8]: Why do not inherit directly from `IPasswordHasher<ApplicationUser>`?

File: src/VirtoCommerce.Platform.Core/Security/Events/UserPasswordChangedEvent.cs [hunk 2]
```diff
@@ -4,11 +4,17 @@ namespace VirtoCommerce.Platform.Core.Security.Events
...
             UserId = userId;
+            HashedPassword = hashedPassword;
         }
 
         public string UserId { get; set; }
+
+        /// <summary>
+        /// Password hash for external hash storage. This provided as workaround until password hash storage would implemented
+        /// </summary>         
+        public string HashedPassword { get; set; }
```
Comment (tatarincev) [inline src/VirtoCommerce.Platform.Core/Security/Events/UserPasswordChangedEvent.cs:18]: Rename to "CustomPasswordHash"

File: src/VirtoCommerce.Platform.Core/Security/IUserPasswordHasher.cs (hunk 1 above)
Comment (akak1977) [reply src/VirtoCommerce.Platform.Core/Security/IUserPasswordHasher.cs]: Ok.

File: src/VirtoCommerce.Platform.Core/Security/Events/UserPasswordChangedEvent.cs (hunk 2 above)
Comment (akak1977) [reply src/VirtoCommerce.Platform.Core/Security/Events/UserPasswordChangedEvent.cs]: Ok.

**Pull request merged**